### PR TITLE
Swipe card and liking

### PR DESCRIPTION
**Pull request by matthewdking**
relates #28 

- swipe functionality using redux-swing and swing modules
- implement action to add career to liked
- updates query for getting unliked careers only

_When running this locally make sure you create a user first_ :+1: 

There are probably a few mistakes so in Marios words _go at it_ :hear_no_evil:

### CODE REVIEW

**Comment by finnhodgkin** [comment]
@matthewdking I think there are some merge conflicts here. Please solve.

**Comment by matthewdking** [comment]
@finnhodgkin :snake:

File: src/Routes.js [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
 import { Route, Switch, BrowserRouter } from 'react-router-dom';
+import styled from 'styled-components';
 import Careers from './containers/Careers.js';
 import LikedCareers from './containers/LikedCareers.js';
 import CareerDetails from './containers/CareerDetails.js';
 import NavBar from './components/NavBar';
 
+const AppWrapper = styled.div`
+  overflow: hidden;
+`;
```
Comment (veredrec) [inline src/Routes.js]: Just a question - what does the wrapper `overflow:hidden` do?

File: src/components/Card.js [hunk 2]
```diff
@@ -2,14 +2,8 @@ import React from 'react';
...
-  background-color: white;
-  height: 70%;
-  margin: 0 2rem;
-  overflow: auto;
-  border-width: thin;
-  border-radius: 1rem;
-  box-shadow: 0 0.7rem 1.2rem rgba(0, 0, 0, 0.09),
-    0 0.4rem 0.4rem rgba(0, 0, 0, 0.16);
+  height: 100%;
+  padding-top: 2%;
```
Comment (veredrec) [inline src/components/Card.js]: Another question about using `rem` and `%` - is the idea here to use `%` for height and `rem` for the rest?

File: src/containers/Careers.js [hunk 3]
```diff
@@ -15,25 +19,122 @@ const CareersContainer = styled.div`
...
+            swipeCount++; // someone please suggest a better way to do this
+            this.likeCareer(e);
+          }}
+          config={{
+            allowedDirections: [Direction.LEFT, Direction.RIGHT],
+            maxRotation: 5,
+            minThrowOutDistance: 1000,
+            maxThrowOutDistance: 1400,
+            throwOutConfidence: (xOffset, yOffset, element) => {
+              const xConfidence = Math.min(
```
Comment (veredrec) [inline src/containers/Careers.js]: Why have you decided to call it `xConfidence` and `yConfidence`? I'm not sure if I understand the naming... :flushed:

File: src/Routes.js (hunk 1 above)
Comment (matthewdking) [reply src/Routes.js]: It was to hide the cards after they had been thrown out the stack

File: src/containers/Careers.js (hunk 3 above)
Comment (matthewdking) [reply src/containers/Careers.js]: This was taken from the swing module docs so it was easier to throw the card and the user didn't need to drag it really close to the edge of the screen

File: src/components/Card.js (hunk 2 above)
Comment (matthewdking) [reply src/components/Card.js]: I'm not really sure to be honest :man_shrugging:

File: src/Routes.js (hunk 1 above)
Comment (veredrec) [reply src/Routes.js]: :+1:

File: src/components/Card.js (hunk 2 above)
Comment (veredrec) [reply src/components/Card.js]: Do we want to style everything with one of them only? I would say that `%` makes sense for height and width, but maybe for all the rest we should use `rem` :straight_ruler:

Comment (matthewdking) [reply src/components/Card.js]: I think it depends on if we want things to be a set height. Eg the navbar would look fine with one percentage in mobile but in desktop the percentage would look silly so rem makes sense for this. I will have a look at what I'm styling here and see what makes sense but maybe its just dependant on the specific case

Comment (veredrec) [reply src/components/Card.js]: Sure, should I wait with approving the PR or go for it?  :running_woman:

Comment (matthewdking) [reply src/components/Card.js]: I just deleted that line and things didn't change haha so who knows what that was for :man_shrugging:

Comment (veredrec) [reply src/components/Card.js]: haha... approving :smile:

File: public/index.html [hunk 4]
```diff
@@ -12,6 +12,7 @@
     <link href="https://fonts.googleapis.com/icon?family=Material+Icons"
       rel="stylesheet">
     <link href="https://fonts.googleapis.com/css?family=Heebo" rel="stylesheet">
+    <link rel="stylesheet" href="./card.css">
```
Comment (finnhodgkin) [inline public/index.html]: Why is this necessary? Shouldn't this be styled components? Even if you're forced to have styled classes, keep them to the SC reset file rather than a css file.

File: public/card.css [hunk 5]
```diff
@@ -0,0 +1,18 @@
+.cardwrapper {
```
Comment (finnhodgkin) [inline public/card.css]: As discussed in person, remove this file and replace it with a `styled` div.

File: public/card.css [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  background-color: white;
+  height: 100%;
+  width: 100%;
+  position: absolute;
+  border-width: thin;
+  border-radius: 1rem;
+}
+
+.cardwrapper:last-child {
```
Comment (finnhodgkin) [inline public/card.css]: Last and first child can be combined:

```css
&:last-child, &:first-child {
box-shadow: 0 0.22rem 0.44rem rgba(0, 0, 0, 0.09),
  0 0.22rem 0.44rem rgba(0, 0, 0, 0.13);
}
```

File: src/components/LikeAndDislikeButtons.js [hunk 7]
```diff
@@ -30,14 +30,24 @@ const DislikedIcon = styled(Icon)`
...
 
-const LikeAndDislikeButtons = () => {
+const LikeAndDislikeButtons = ({ throwCard }) => {
   return (
     <ButtonContainer>
       <IconWrapper>
-        <DislikedIcon className="material-icons">clear</DislikedIcon>
+        <DislikedIcon
+          className="material-icons"
+          onClick={() => throwCard('left')}
```
Comment (finnhodgkin) [inline src/components/LikeAndDislikeButtons.js]: Would be really great if you could apply an animation to the card on click so it looks like it's being thrown out, rather than just disappearing.

File: src/components/LikeAndDislikeButtons.js [hunk 8]
```diff
@@ -30,14 +30,24 @@ const DislikedIcon = styled(Icon)`
...
+        >
+          clear
+        </DislikedIcon>
       </IconWrapper>
       <IconWrapper>
-        <LikedIcon className="material-icons">done</LikedIcon>
+        <LikedIcon
+          className="material-icons"
+          onClick={() => throwCard('right')}
```
Comment (finnhodgkin) [inline src/components/LikeAndDislikeButtons.js]: Same as above.

File: src/containers/Careers.js [hunk 9]
```diff
@@ -15,25 +19,122 @@ const CareersContainer = styled.div`
...
+const SwingWrapper = styled.div`
+  position: relative;
+  height: 70%;
+  width: 80%;
+  margin: 0 2rem;
+  z-index: 0;
+  overflow: visible;
+`;
+
+let swipeCount = 1;
```
Comment (finnhodgkin) [inline src/containers/Careers.js]: I really feel like this should be in state. Kinda the point of using state in the first place.

File: src/containers/Careers.js [hunk 10]
```diff
@@ -15,25 +19,122 @@ const CareersContainer = styled.div`
...
+    if (e.throwDirection.toString() === 'Symbol(RIGHT)') {
+      return this.props.likeCareerAction({
+        user_id: 1,
+        career_id: Number(e.target.id.slice(4)),
+      });
+    }
+  };
+
+  throwCard = direction => {
+    const cardRef = Object.keys(this.refs.stack.refs)[
```
Comment (finnhodgkin) [inline src/containers/Careers.js]: As discussed in person, clear up the variable naming so it's a bit clearer what's happening here and you don't make duplicate calls to Object.keys, etc.

File: src/containers/Careers.js [hunk 11]
```diff
@@ -15,25 +19,122 @@ const CareersContainer = styled.div`
...
+    return card.throwOut(0.3, 0.1, Swing.DIRECTION.LEFT);
+  };
+
   render() {
     return (
       <CareersContainer>
-        <Card />
-        <LikeAndDislikeButtons />
+        <SwingWrapper>{this.swingShow()}</SwingWrapper>
+        <LikeAndDislikeButtons throwCard={this.throwCard.bind(this)} />
```
Comment (finnhodgkin) [inline src/containers/Careers.js]: Remove the like and dislike buttons when there are no more cards remaining. Happy for this to be in the next PR.

File: database/queries/get_all_careers.js [hunk 12]
```diff
@@ -1,8 +1,8 @@
 const dbConnection = require('../db_connection');
 
 const getAllCareers = callback => {
-  const allCareersQuery = 'SELECT * FROM careers;';
-  dbConnection.query(allCareersQuery, (error, response) => {
+  const allCareersQuery = `SELECT * FROM careers WHERE careers.id NOT IN (SELECT career_id FROM users_careers WHERE users_careers.user_id = $1);`;
```
[the remark this answers is not in the document]
Comment (matthewdking) [reply database/queries/get_all_careers.js]: :relieved: Thanks man

File: src/containers/Careers.js (hunk 11 above)
Comment (matthewdking) [reply src/containers/Careers.js]: Watch this space :eyes:

File: public/index.html (hunk 4 above)
Comment (matthewdking) [reply public/index.html]: Removed :wastebasket:

File: src/containers/Careers.js (hunk 9 above)
Comment (matthewdking) [reply src/containers/Careers.js]: Done :+1:

File: src/components/LikeAndDislikeButtons.js (hunk 7 above)
Comment (matthewdking) [reply src/components/LikeAndDislikeButtons.js]: Having researched this I don't think it is possible

Comment (finnhodgkin) [reply src/components/LikeAndDislikeButtons.js]: Yeah maybe raise it in a `nice to have` issue :+1:
